src/hotspot/share/opto/vectornode.hpp

```diff
@@ -73,6 +73,15 @@ class VectorNode : public TypeNode {
     return type()->ideal_reg();
   }
 
+  virtual uint hash() const {
+    if (is_commutative_vector_operation() && !is_predicated_vector()) {
```

Comment:
You should either put the comment here again, or even better move out the condition:
`is_commutative_vector_operation() && !is_predicated_vector()`